data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessorIT.java

```diff
@@ -218,16 +221,96 @@ void aggregateWithPutAllActionAndCondition() throws InterruptedException {
+
+        Map<String, Object> expectedEventMap = new HashMap<>(getEventMap(1));
+        expectedEventMap.put(CountAggregateAction.COUNTKEY, NUM_THREADS * NUM_EVENTS_PER_BATCH);
+
+        Record<Event> record = (Record<Event>)eventBatch.toArray()[0];
+        assertThat(expectedEventMap, equalTo(record.getData().toMap()));
+    }
+
+    @RepeatedTest(value = 2)
+    void aggregateWithCountAggregateActionWithCondition() throws InterruptedException {
+        aggregateAction = new CountAggregateAction();
+        when(pluginFactory.loadPlugin(eq(AggregateAction.class), any(PluginSetting.class)))
+                .thenReturn(aggregateAction);
+        final String condition = "/firstRandomNumber < 100";
+        when(aggregateProcessorConfig.getGroupDuration()).thenReturn(Duration.ofSeconds(GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE));
+        when(aggregateProcessorConfig.getWhenCondition()).thenReturn(condition);
+        int count = 0;
+        eventBatch = getBatchOfEvents(true);
+        Record<Event> rec = (Record<Event>)eventBatch.toArray()[0];
+        for (Record<Event> record: eventBatch) {
+            Event event = record.getData();
+            boolean value = (count % 2 == 0) ? true : false;
+            when(expressionEvaluator.evaluate(condition, event)).thenReturn(value);
+            count++;
+        }
+
+        final AggregateProcessor objectUnderTest = createObjectUnderTest();
+
+        final ExecutorService executorService = Executors.newFixedThreadPool(NUM_THREADS);
+        final CountDownLatch countDownLatch = new CountDownLatch(NUM_THREADS);
+
+        for (int i = 0; i < NUM_THREADS; i++) {
+            executorService.execute(() -> {
+                final List<Record<Event>> recordsOut = (List<Record<Event>>) objectUnderTest.doExecute(eventBatch);
+                countDownLatch.countDown();
+            });
+        }
+        Thread.sleep(GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE * 1000);
+
+        boolean allThreadsFinished = countDownLatch.await(5L, TimeUnit.SECONDS);
+        assertThat(allThreadsFinished, equalTo(true));
+
+        Collection<Record<Event>> results = objectUnderTest.doExecute(new ArrayList<Record<Event>>());
+        assertThat(results.size(), equalTo(1));
+
+        Map<String, Object> expectedEventMap = new HashMap<>(getEventMap(1));
+        expectedEventMap.put(CountAggregateAction.COUNTKEY, NUM_THREADS * NUM_EVENTS_PER_BATCH/2);
+
+        rec = (Record<Event>)results.toArray()[0];
+        assertThat(expectedEventMap, equalTo(rec.getData().toMap()));
+    }
+
+    private List<Record<Event>> getBatchOfEvents(boolean withSameValue) {
         final List<Record<Event>> events = new ArrayList<>();
 
         for (int i = 0; i < NUM_EVENTS_PER_BATCH; i++) {
-            final Map<String, Object> eventMap = getEventMap(i % NUM_UNIQUE_EVENTS_PER_BATCH);
+            final Map<String, Object> eventMap = (withSameValue) ? getEventMap(1) : getEventMap(i % NUM_UNIQUE_EVENTS_PER_BATCH);
```

Comment:
`getEventMap(1)` is using a magic number. What is the meaning of `1`? Perhaps make a constant with a name.